src/app/core/services/label.service.ts

```diff
@@ -76,6 +76,18 @@ export class LabelService {
     return requiredLabels;
   }
 
+    /**
+   * Returns an custom operator which checks if the session had been created
+   * either naturally or using a fix.
+   * If true, returns a call to synchronise the labels in our application
+   * with the remote repository. Else, throw an error.
+   */
+  syncLabels(): UnaryFunction<Observable<boolean>, Observable<any>> {
+    return pipe(
+      map(() => this.synchronizeRemoteLabels())
```

Comment:
There's a regression bug here - the label synchronisation isn't being triggered (try it out on your bug reporting phase repo). We need to use `flatMap` to ensure the inner observable gets subscribed to.

Reply:
Ah I see... Might need to brush up my knowledge with regards to observables and monads / functors